### PR TITLE
Revenue by clients report page UI and common context for all reports.

### PR DESCRIPTION
**Pull request by onkar-saeloun**
## Notion card
https://www.notion.so/saeloun/Create-the-front-end-for-revenue-by-clients-reports-b170ac9a1b2c44d1961fed9393a05a82

## Summary

- Added Report page for revenue by client
- Added filters for the same
- Added single context for all the reports 
- Added validation on the filters

## Preview

https://user-images.githubusercontent.com/104751221/176852468-4b6861a4-a05a-4c78-90c9-9c778fb977ad.mp4

## Type of change

Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Tested Manually on my local machine.

### Checklist:

- [ ] I have manually tested all workflows
- [ ] I have performed a self-review of my own code
- [ ] I have added automated tests for my code

### CODE REVIEW

**Comment by apoorv1316** [comment]
@onkar-saeloun Clear all button is not removing the "this quarter" filter. Please check.

**Review by apoorv1316 — APPROVED** [review]

**Review by harshaanNihal — APPROVED** [review]
Added some nitpicks 

Approving as PR LGTM ✨

File: app/javascript/src/common/CustomDateRangePicker/index.tsx [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+  selectedInput,
+  dateRange
+}) => {
+  const range = (start, end) => {
+    const ans = [];
+    for (let i = start; i <= end; i++) {
+      ans.push(i);
+    }
+    return ans;
+  };
```
Comment (JijoBose) [inline app/javascript/src/common/CustomDateRangePicker/index.tsx:22]: can we replace this block with something like this instead?

```
const range = (start, end) => {
  return Array.from({length: (end - start)}, (v, k) => k + start);
}
```

File: app/javascript/src/components/Reports/Header/index.tsx [hunk 2]
```diff
@@ -24,7 +24,20 @@ const Header = ({
...
+  const getReport = (currentReport) => {
+    switch (currentReport) {
+      case "RevenueByClientReport":
+        return revenueByClientReport;
+      case "TimeEntryReport":
+        return timeEntryReport;
+      default:
+        break;
+    }
+  };
```
[the remark this answers is not in the document]
Comment (onkar-saeloun) [reply app/javascript/src/components/Reports/Header/index.tsx:37]: @harshaanNihal This refactoring is already done in API integration branch will raise separate PR for this.

**Review by JijoBose — APPROVED** [review]

**Pull request merged**